src/lib/background/extension.ts

```diff
@@ -193,10 +192,9 @@ export function Extension() {
             await env.verify();
             await env.boot();
 
-            const { hasOnboarded } = await getLocalValues();
-
-            if (this.exists() && hasOnboarded) {
-                // If profile exists, it means that the extension was restarted.
+            if (this.exists() && !this.profile().password().exists()) {
```

Comment:
this won't work as it creates a profile during onboarding. If you reload the plugin during onboarding it will end up showing this value as the password is gone but the user has no idea what the randomly generated password is and will be locked out